django/db/models/functions/text.py

```diff
@@ -235,6 +236,54 @@ class RTrim(Transform):
     lookup_name = 'rtrim'
 
 
+class SHA1(MD5):
+    function = 'SHA1'
+    lookup_name = 'sha1'
+
+    def as_postgresql(self, compiler, connection, **extra_content):
+        # digest() function requires the pgcrypto extension.
+        return super().as_sql(
+            compiler,
+            connection,
+            template="ENCODE(DIGEST(%(expressions)s, '%(function)s'), 'hex')",
+            function=self.function.lower(),
+            **extra_content,
+        )
+
+
+class SHA224(SHA1):
+    function = 'SHA224'
+    lookup_name = 'sha224'
+
+    def as_mysql(self, compiler, connection, **extra_content):
+        return super().as_sql(
+            compiler,
+            connection,
+            template='SHA2(%%(expressions)s, %s)' % self.function[3:],
+            **extra_content,
+        )
+
+    def as_oracle(self, compiler, connection, **extra_context):
+        if self.function != 'SHA224':
```

Comment:
Again, the inheritance chain looks questionable to me if this is needed.

Reply:
Yeah, agreed.